### ISSUE TITLE
Decline in AIS Performance After Fine-Tuning uSAM on Custom Microscopy Dataset

### ISSUE THREAD

**Issue by ChangJX521** [computational-cell-analytics/micro-sam#920]
# **Decline in AIS Performance After Fine-Tuning uSAM on Custom Microscopy Dataset**  

I'm attempting to fine-tune uSAM on my own dataset following the instructions in [sam_finetuning.ipynb](https://github.com/computational-cell-analytics/micro-sam/blob/master/notebooks/sam_finetuning.ipynb), but the segmentation results have worsened after fine-tuning.  

## **Steps to Reproduce**  

1. Training on microscopy cell images  
2. Using the `vit_b_lm` model type with automatic instance segmentation  
3. Training for 10 epochs with a batch size of 2 and a patch size of (512, 512)  

<img width="956" alt="Training process" src="https://github.com/user-attachments/assets/4472bc4d-30cd-428d-b566-f5fd914d7617" />  

Despite keeping the parameters largely unchanged, the AIS performance has deteriorated after fine-tuning.  

<img width="960" alt="Fine-tuning result" src="https://github.com/user-attachments/assets/0392e49e-1ef1-48bd-91a7-b884d271e3aa" />  

## **Results Before Fine-Tuning**  

The following result was obtained using `vit_b_lm`, following the instructions in [automatic_segmentation.ipynb](https://github.com/computational-cell-analytics/micro-sam/blob/master/notebooks/automatic_segmentation.ipynb).  

![Before fine-tuning](https://github.com/user-attachments/assets/595377b9-5bac-4acc-bced-be723d92a68f)  

## **Results After Fine-Tuning**  

![After fine-tuning](https://github.com/user-attachments/assets/36bd5f6d-c33f-46e5-a64e-47f177ebbc14)  

As shown in the images, fine-tuning has led to a decline in segmentation quality. Some adjacent cells are now incorrectly merged into a single instance, background stains are mistakenly identified as cells, and some cells are not detected at all.

I experimented with increasing the number of epochs, adjusting the learning rate, and modifying `n_objects_per_batch`, but in some cases, the results worsened even further—sometimes to the extent that nothing was recognized at all.  

![Failed segmentation](https://github.com/user-attachments/assets/6c2ecbdb-d041-4694-9581-82143c8b5158)  

## **Dataset**  

The dataset consists of bright-field microscopy images of cells seeded on the surface of an SPR chip. Each image is in RGB format with a resolution of **3072 × 2048 pixels**, totaling approximately **40 images**. The number of cells per image varies between **20 and 150**.  

To minimize the impact of dataset quality, I selected images with well-annotated cells and relatively uniform cell distribution for training. However, the segmentation issues persist.  

![Sample dataset image](https://github.com/user-attachments/assets/d1e90fd0-5b5c-4e7e-a8c0-bba487b27441)  

## **Conclusion**  

Despite closely following the fine-tuning guidelines and experimenting with different hyperparameters, the segmentation performance has not improved and, in some cases, has worsened significantly. I would greatly appreciate any insights or suggestions on what might be causing these issues and how to improve the fine-tuning process.  

Thank you in advance for your time and help!

**Comment by anwai98** [comment]
Hi @ChangJX521,

First of all, thanks for your interest in `micro-sam`, and the detailed break down of your observation.

I have a few questions, for us to understand your data a bit better?
- What are the images you presented, and the modality of the images? (if possible, it would be great if you could send us some sample images and corresponding labels (if you have annotations) for us to understand this better)
- What is the spatial shape of your images? (from the crops you sent used for training, I am assuming the images are quite large)
- When using automatic segmentation, and assuming the images are larger than (1024, 1024), are you using automatic segmentation with tiling?

I have a few suspicions and a probable solution which will help us. I will wait for you to answer the questions first! ;)

**Comment by ChangJX521** [comment]
Thank you for your response! I appreciate your time and support. Below are my answers to your questions:

The images are bright-field microscopy images of cells seeded on an SPR chip surface. yes, i**t's 3072 × 2048**,   RGB I notice there is a param of tile_shape in the automatic_instance_segmentation function, which in my case in set to None, the same as [sam_finetuning.ipynb](https://github.com/computational-cell-analytics/micro-sam/blob/master/notebooks/sam_finetuning.ipynb).

**Comment by ChangJX521** [comment]
@anwai98 
this is an example of my data, which I use napari to annotate
https://drive.google.com/drive/folders/12E90j-cSTkOPygwpvXKfgjzf1rSLpd54?usp=share_link

**Comment by ChangJX521** [comment]
add  tile_shape = (512, 512) could solve the problem, thank you❤️
